Feat : channelUrl -> channelId

voka opened the pull request:
- video 저장로직 수정 (channelId로 바뀐 부분)
- 사용하지 않는 VideoQueryService 파일 삭제 
- VideoRequest에 Example 추가
- 사용하지 않는 ItemQueryService 파일 및 관련 파일 삭제

voka committed:
Feat: 영상에 대한 시간(TimeFrame) CRUD API 기능 추가

voka committed:
Update: 시간 생성시 요청 시간과 동일한 시간에 생성된 객체가 있는지 검사하는 로직 추가

voka committed:
Feat: 시간에 상품과 쇼핑몰 링크 추가하는 API 기능 추가

- 같은 시간에 동일한 상품 및 동일한 쇼핑몰 링크가 있을경우 Error 발생하게 처리

voka committed:
Update: channel 엔티티의 필드 channel_url을 channel_id로 변경

voka committed:
Update: channelUrl -> channelId로 바꾸면서 Video 엔티티 저장 로직 수정

voka committed:
Update: 사용하지 않는 파일 삭제, Request에 Example 추가

hgoolee commented on src/main/java/com/clovi/timeShopItem/TimeShopItemController.java:
```diff
@@ -0,0 +1,42 @@
+    public ResponseEntity createTimeShopItem(@PathVariable(name = "video_id") Long videoId, @AuthMember Member member, @PathVariable(name = "time_frame_id") Long timeFrameId
+            , @PathVariable(name = "item_id") Long itemId, @PathVariable(name = "shop_item_id") Long shopItemId){
+        SavedId savedId = new SavedId(timeShopItemService.create(member,timeFrameId,itemId,shopItemId));
+        return ResponseEntity.created(
```
```java
String[] list = {"/api/v1/videos", String.valueOf(videoId), "timeframes", String.valueOf(timeFrameId)};
URI.create(String.join("/", list)).body( ...
```

이렇게 쓰면 좀 더 깔끔하지 않을까 싶습니다.

voka replied:
반영완료.

hgoolee commented on src/main/java/com/clovi/timeShopItem/TimeShopItemController.java:
```diff
@@ -0,0 +1,42 @@
+            @ApiResponse(responseCode = "201", description = "Success create", content = @Content(schema = @Schema(implementation = SavedId.class)))
+    })
+    public ResponseEntity createTimeShopItem(@PathVariable(name = "video_id") Long videoId, @AuthMember Member member, @PathVariable(name = "time_frame_id") Long timeFrameId
+            , @PathVariable(name = "item_id") Long itemId, @PathVariable(name = "shop_item_id") Long shopItemId){
```
`videoId`, `timeFrameId`, `itemId`, `shopItemId` 모두 null 또는 "" 또는 " "이 오면 안 되므로 `@NotBlank`로 유효성 검사를 해야 할 것 같습니다.

voka replied:
@PathVariable은 기본적으로 required 옵션이 true라 @NotBlank 같은 유효성 검사 어노테이션을 안붙여도 되지 않을까 생각합니다.

hgoolee replied:
오케이 그러면 괜찮을 거 같네요ㅎㅎ

hgoolee commented on src/main/java/com/clovi/timeframe/TimeFrameController.java:
```diff
@@ -0,0 +1,86 @@
+    @Operation(summary = "Find all timeFrame", description = "Find all timeFrame by video ID.", responses = {
+            @ApiResponse(responseCode = "200", description = "Success Find timeFrame List", content = @Content(schema = @Schema(implementation = TimeFrameResponseDto.class)))
+    })
+    public ResponseEntity findAllTimeFramesByVideoId(@PathVariable(name = "video_id") Long video_id){
```
변수명은 camel case를 사용 중이므로 `video_id`를 `videoId`로 변경 부탁드립니다.

voka replied:
반영완료.

hgoolee commented on src/main/java/com/clovi/timeframe/TimeFrameService.java:
```diff
@@ -1,8 +1,83 @@
+    private final TimeFrameRepository timeFrameRepository;
+    private final VideoRepository videoRepository;
+
+    public List<TimeFrameResponseDto> getTimeFrameListByVideoId(Long video_id) {
```
마찬가지로 여기에서도 `video_id`를 `videoId`로 변경 부탁드립니다.

voka replied:
반영완료.

hgoolee commented on src/main/java/com/clovi/timeframe/TimeFrameService.java:
```diff
@@ -1,8 +1,83 @@
+    public List<TimeFrameResponseDto> getTimeFrameListByVideoId(Long video_id) {
+        List<TimeFrame> timeFrames = timeFrameRepository.findAllByVideoIdAndDeletedIsFalse(video_id);
+        List<TimeFrameResponseDto> result = timeFrames.stream().map(TimeFrameResponseDto::new).collect(Collectors.toList());
+        if(result.isEmpty() == false) {
```
`if(!result.isEmpty())`가 더 직관적일 것 같습니다.

voka replied:
반영완료.

hgoolee commented on src/main/java/com/clovi/timeframe/TimeFrame.java:
```diff
@@ -54,6 +57,17 @@ public TimeFrame(Long capturePoint, Model model, Video video) {
     this.video = video;
   }
 
+  public TimeFrame(TimeFrameCreateRequest timeFrameCreateRequest, Video video, Long id) {
```
Entity의 생성자 안에서 DTO를 사용하게 되면 Request의 형식에 변화가 있을 때마다 Entity 코드를 수정해야 하므로 Entity가 DTO에 의존적이게 되고, 이러한 생성자를 계속해서 추가하게 되면 데이터베이스 테이블의 매핑 연관관계를 정의하는 Entity의 원래 성격에 맞지 않는다는 생각이 듭니다.
그래서 DTO 파일에서 toEntity라는 메서드를 추가하여 거기서 DTO -> Entity 변환을 수행하면 더 나을 것 같습니다.
`TimeFrameCreateRequest.java` 파일에 예시를 첨부하겠습니다.

voka replied:
ㅇㅎ 그러면 다른 Request 들도 함께 반영하는게 좋겠네요 다른 브랜치 하나파서 일괄적으로 처리할게요~

hgoolee replied:
네넵 다른 request들도 다 그렇게 하면 될 것 같아요:)

hgoolee commented on src/main/java/com/clovi/timeframe/dto/request/TimeFrameCreateRequest.java:
```diff
@@ -0,0 +1,28 @@
+
+    public Long getTime() {
+        return TimeFormatUtils.StringTimeToLong(this.time);
+    }
```
위에서 언급했던 예시입니다.

```java
public TimeFrame toEntity(Video video, Long id) {
  return TimeFrame.builder()
          .capturePoint(time)
          .video(video)
          .createBy(id)
          .lastModifiedBy(id)
          .build();
```

`TimeFrame.java`에서 생성자 위에 `@Builder`를 사용해야 합니다.

hgoolee reviewed (commented):
코드 리뷰 완료했습니다:)

voka committed:
Update: video_id -> youtube_video_id로 변경

voka committed:
Update: 에러메세지 변경
- NotFoundException 대신 BadRequestExcception 을 사용하는 것으로 일괄 변경

voka committed:
Refactor: https://github.com/Soma13-TigerSpoon/CLOVI-server/pull/75

voka committed:
Fix: Resolve merge conflict

voka committed:
Fix: Resolve merge Conflict
- LoggingConfig 파일 이름 변경
- 사용하지 않는 저장 로직에 사용되는 파일
	- ItemQueryService 삭제
	- TimeShopItemRequest 삭제
